### PR TITLE
First components of activePage

### PR DESCRIPTION
**Pull request by Nepec**
# Description
Adding the following components to the activePage section:
- **ActiveTemplates** - accordion in both a simple and nested version. The simple is thought for the student view and contains simple VM instances, while the nested version is thought for the professor view and is divided into courses->termplates->VMs
- **TemplateHeader** - title of each accordion displaying the number of active VM/the number of templates and a "delete all" button. The button has a responsive behavior as it is substituted by a "more" icon when the page shrinks below the `md` breakpoint
- **ActiveLanding** - a generic temporary wrapper to better display the accordion on the local barebones website. It should be placed in the `<Route path="/active"></Route>` comopnent in `App.tsx`

# Notes
The data structures contained in `./activePage/utils/data` are temporary and are used only to showcase the components.

As pointed out in the Slack channel there are issues in the correct rendering of components on Storybook: some Tailwind properties conflict with some Ant properties. I've managed to work around most of them, however there are still issues with:
- background colors not being displayed
- responsive behavior not working: in the current iteration both the "Delete All" button and the "more" icon will be visible at all times

### CODE REVIEW

File: frontend/src/components/activePage/utils/data.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+export interface IVM {
```
Comment (GabriFila) [inline frontend/src/components/activePage/utils/data.tsx:1]: Since you use this file inside a single component, decelare the needed interface there and if needed, import them in the stories

File: frontend/src/components/activePage/ActiveTemplates/ActiveTemplates.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  nested: boolean;
+}
+
+const ActiveTemplates: FC<IActiveTemplatesProps> = ({ ...props }) => {
+  const { templates, courses, destroyAll, nested } = props;
+  const { Panel } = Collapse;
+  return (
+    <Collapse
+      accordion
+      className="bg-gray-200 border-0 mx-10"
```
Comment (GabriFila) [inline frontend/src/components/activePage/ActiveTemplates/ActiveTemplates.tsx:20]: If you look at the storybook, when in mobile view, the `mx-10` is too much. I would reduce it in that case

File: frontend/src/components/activePage/ActiveTemplates/ActiveTemplates.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import { FC } from 'react';
+import { Collapse } from 'antd';
+import { CaretRightOutlined } from '@ant-design/icons';
+import TemplateHeader from '../TemplateHeader/TemplateHeader';
+import { ITemplate, ICourse } from '../utils/data';
+
+export interface IActiveTemplatesProps {
+  courses: Array<ICourse>;
```
Comment (GabriFila) [inline frontend/src/components/activePage/ActiveTemplates/ActiveTemplates.tsx:8]: Change courses to workspaces

File: frontend/src/components/activePage/ActiveTemplates/ActiveTemplates.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import { Title, Description, Stories } from '@storybook/addon-docs/blocks';
+import { someKeysOf } from '../../../utils';
+import { courses, templates } from '../utils/data';
+
+export default {
+  title: 'Components/ActivePage/ActiveTemplates',
+  component: ActiveTemplates,
+  argTypes: { destroyAll: { action: 'clicked' } },
+  parameters: {
+    docs: {
```
Comment (GabriFila) [inline frontend/src/components/activePage/ActiveTemplates/ActiveTemplates.stories.tsx:13]: It is great you added the docs to the storry!

If you want and when you have time, you could make another PR where you add instructions in the guidelines on how to do it. The docs on the stories would still be optional, but if someone would like to add them, he/she would already know how to do it

**Review by GabriFila — CHANGES_REQUESTED** [review]

File: frontend/src/components/activePage/ActiveLanding/ActiveLanding.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import { FC, useState } from 'react';
+import { Col, Switch } from 'antd';
+import WorkspacesTable from '../Workspaces/WorkspacesTable';
+import TemplatesTable from '../Templates/TemplatesTable';
+import Box from '../../common/Box';
+
+export interface IWorkspace {
```
Comment (GabriFila) [inline frontend/src/components/activePage/ActiveLanding/ActiveLanding.tsx:7]: This data should be placed in the story. for the moment pass it as a prop. Later it may become a context

File: frontend/src/components/activePage/ActiveLanding/ActiveLanding.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        <>
+          <Title />
+          <Description />
+        </>;
+      },
+    },
+  },
+  decorators: [
+    (Story: any) => (
+      <div style={{ margin: '2rem', height: '500px' }}>
```
Comment (GabriFila) [inline frontend/src/components/activePage/ActiveLanding/ActiveLanding.stories.tsx:20]: Why do you use this decorator? Also, on mobile/tablet it doesn't fit well

File: frontend/src/components/activePage/ActiveLanding/ActiveLanding.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+const ActiveLanding: FC = () => {
+  const [managerView, setManagerView] = useState(false);
+  return (
+    <Box
+      header={{
+        size: 'small',
+        center: (
+          <div className="flex justify-end ml-auto">
+            <Switch
```
Comment (GabriFila) [inline frontend/src/components/activePage/ActiveLanding/ActiveLanding.tsx:69]: Could you please look for an icon that explain a bit more the concept of manager/user?
Also I would add here the check to hide the switch if the user is never manager.

File: frontend/src/components/activePage/RowHeading/RowHeading.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+          }}
+        >
+          Destory All
+        </Button>
+        <Dropdown
+          overlay={
+            <Menu>
+              <Menu.Item icon={<DeleteOutlined className="text-lg" />} danger>
+                {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+                <a
```
Comment (GabriFila) [inline frontend/src/components/activePage/RowHeading/RowHeading.tsx:47]: Why did you use an ancora tag for the `DestroyAll`?

File: frontend/src/components/activePage/Workspaces/WorkspacesTable.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        ),
+        // eslint-disable-next-line react/no-multi-comp
+        expandedRowRender: record => (
+          <TemplatesTable
+            templates={record.templates}
+            destroyAll={destroyAll}
+          />
+        ),
+        expandRowByClick: true,
+        indentSize: 100,
```
Comment (GabriFila) [inline frontend/src/components/activePage/Workspaces/WorkspacesTable.tsx:45]: I don't think it is strictly related to this line. But check the storybook on mobile, because the workspaces/templates sections become very small and the `...` menu gets pushed on a new line

File: frontend/src/components/activePage/Templates/TemplatesTable.tsx [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    >
+      <Column
+        title="Templates"
+        dataIndex="name"
+        key="key"
+        render={(text, record: ITemplate) => {
+          return (
+            <RowHeading
+              text={record.name}
+              nActive={record.nActiveMachines}
```
Comment (GabriFila) [inline frontend/src/components/activePage/Templates/TemplatesTable.tsx:59]: Use `nActiveInstances` instead of `nActiveMachines`

File: frontend/src/components/activePage/ActiveLanding/ActiveLanding.tsx (hunk 5 above)
Comment (Nepec) [reply frontend/src/components/activePage/ActiveLanding/ActiveLanding.tsx:7]: I'm not sure how to organize my data since I need its interfaces to build the `IActiveLandingProps` interface. Should I declare the `Itemplate` and `IWorkspaces` interfaces here and move the actual data structures in the story (which would then import the interfaces from this file)?

File: frontend/src/components/activePage/Workspaces/WorkspacesTable.tsx (hunk 9 above)
Comment (Nepec) [reply frontend/src/components/activePage/Workspaces/WorkspacesTable.tsx:45]: As far as the mobile view is concerned I haven't really focused on that screen size as this frontend is Desktop oriented and the mobile view is just too narrow. I've focused around the `lg` and `md` breakpoints thinking of when a user splits his screen between the Crownlabs site and another window.
Regardless, I want to add ellipses to all the template and workspace titles which should mitigate the problem also on the narrowest of views

Also I should remove the `indentSize` line as it does not work at all.

File: frontend/src/components/activePage/ActiveLanding/ActiveLanding.stories.tsx (hunk 6 above)
Comment (Nepec) [reply frontend/src/components/activePage/ActiveLanding/ActiveLanding.stories.tsx:20]: Just to center the view in the Docs a bit and have a high-enough window to show the entire table without scrolling. I can remove the margin as it actually looks ok without.

Comment (GabriFila) [reply frontend/src/components/activePage/ActiveLanding/ActiveLanding.stories.tsx:20]: I would remove the margins

File: frontend/src/components/activePage/ActiveLanding/ActiveLanding.tsx (hunk 5 above)
Comment (GabriFila) [reply frontend/src/components/activePage/ActiveLanding/ActiveLanding.tsx:7]: You can export the interfaces. If they are used in a single component declare them there, instead if they are used in multiple components, it is better to declare them in a separate file (still in you working folder).
Then import the interfaces in the stories where you need them.

The point is that the component should have no hardcoded values (unless those that will be actually hardcoded when used, e.g. a particular color/setting). The data you wrote is dynamic from the perspective of the component, so if you want to use it make it a prop (or a context with the necessary decorators in the stories)

**Review by GabriFila — APPROVED** [review]

File: frontend/src/components/activePage/NestedTables/NestedTables.tsx [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+  const accordion = (expanded: boolean, record: rowType) => {
+    isManager
+      ? expanded
+        ? setExpandedID([(record as IWorkspace).key])
+        : setExpandedID([''])
+      : expanded
+      ? setExpandedID([(record as ITemplate).key])
+      : setExpandedID(['']);
+  };
```
Comment (QcFe) [inline frontend/src/components/activePage/NestedTables/NestedTables.tsx:44]:
```suggestion
  const accordion = (expanded: boolean, record: rowType) => {
    const expId = !expanded ? '' 
      : (isManager ? (record as IWorkspace).key : (record as ITemplate).key);
    setExpandedID([expId]);
  };
```
This part seems a bit redundant and not too easy to be read. 
I tried to make it cleaner but I don't know if I succeeded. You're totally free to reject this suggestion (especially if it's not correct 😂).

File: frontend/src/components/activePage/ActiveLanding/ViewModeButton/ViewModeButton.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import { FC, Dispatch, SetStateAction } from 'react';
+import { Radio } from 'antd';
+
+export interface IViewModeButtonProps {
+  textUsr: string;
+  textMngr: string;
```
Comment (QcFe) [inline frontend/src/components/activePage/ActiveLanding/ViewModeButton/ViewModeButton.tsx:6]: Do we have any actual need to have these passed from outside? 
(I guess I'm missing something as my usual 🥴)

**Review by QcFe — APPROVED** [review]
A couple of minor things but it's good for me!

File: frontend/src/components/activePage/NestedTables/NestedTables.tsx (hunk 11 above)
Comment (Nepec) [reply frontend/src/components/activePage/NestedTables/NestedTables.tsx:44]: Oh yes, much better thanks!

File: frontend/src/components/activePage/ActiveLanding/ViewModeButton/ViewModeButton.tsx (hunk 12 above)
Comment (Nepec) [reply frontend/src/components/activePage/ActiveLanding/ViewModeButton/ViewModeButton.tsx:6]: Not really, I just thought it would be neat to be able to change the text through a prop.
